src/haxeLanguageServer/features/completion/CompletionFeature.hx

```diff
@@ -280,7 +280,22 @@ class CompletionFeature {
                         case _: field.name;
                     }
                 },
-                range: data.replaceRange
+                range: {
+                    if (data.replaceRange == null){
```

Comment:
`data.completionPosition.toRange()`? ;)

Reply:
Also, is it just that it can't deal with null fields? `range` _should_ be optional. In that case, maybe just only set `range` when we have a `replaceRange`?

Reply:
@Gama11 are you sure it should be optional?

I don't see it in the [specs](https://microsoft.github.io/language-server-protocol/specification#textedit) which have been followed in [languageserver-types (rust)](https://github.com/autozimu/languageserver-types/blob/master/src/lib.rs#L257).

Completion is broken for haxe on nvim when range is not set (fails to deserialize), I had to cherry pick @jdonaldson's commits to make it work.

Reply:
You may be right. So VSCode probably happens to be fine with it anyway, but the spec not so much.